server/src/main/java/org/opensearch/gateway/BaseGatewayShardAllocator.java

```diff
@@ -78,6 +81,19 @@ public void allocateUnassigned(
         executeDecision(shardRouting, allocateUnassignedDecision, allocation, unassignedAllocationHandler);
     }
 
+    protected void allocateUnassignedBatchOnTimeout(List<ShardRouting> shardRoutings, RoutingAllocation allocation, boolean primary) {
```

Comment:
Just to confirm that the reroute would be called even if the decision is Throttle?

Reply:
Yes - https://github.com/opensearch-project/OpenSearch/pull/14848#discussion_r1687999886